src/MOI/MOI_wrapper.jl

```diff
@@ -2842,6 +2842,11 @@ function _check_moi_callback(model::Optimizer)
 end
 
 function _optimize!(model)
+    if _check_moi_callback(model) || model.has_generic_callback
+        println("Callback detected, set 'Threads' to '1' for thread safety")
+        MOI.set(model, MOI.RawOptimizerAttribute("Threads"), 1)
+    end
```

Comment:
My suggestion was to set `Threads` when you set the callback, not here in optimize. Then if the user sets Threads afterwards, they can deal with the consequences.

Reply:
Got it!  We made a new commit as suggested.  Thank you for the clarification.

Reply:
We reverted the commit due to segfault in Julia 1.6.7.   COPT does not currently support changing parameters in callbacks,  so the suggested approach from CPLEX.jl is currently not applicable.  As a work-around,  we reverted the change back to the previous solution, which sets the number of threads in the MOI interface for models with callbacks.